app/src/main/java/com/infomaniak/mail/ui/main/menuDrawer/MenuDrawerAdapter.kt

```diff
@@ -219,12 +262,18 @@ class MenuDrawerAdapter @Inject constructor() : ListAdapter<Any, MenuDrawerViewH
                 onFolderClicked = callbacks.onFolderClicked,
                 onCollapseChildrenClicked = callbacks.onCollapseChildrenClicked,
             )
+            is ActionsHeaderViewHolder -> holder.displayActionsHeader(
+                binding = this as ItemMenuDrawerActionsHeaderBinding,
+                onActionsHeaderClicked = callbacks.onActionsHeaderClicked,
+            )
+            is ActionViewHolder -> holder.displayAction(
+                action = item as MenuDrawerAction,
+                binding = this as ItemMenuDrawerActionBinding,
+                onActionClicked = callbacks.onActionClicked,
+            )
```

Comment:
Done.
I find it way less elegant, since now the casts are done in the middle of the code, and not neatly separated in parameters.

Reply:
I feel like it should be possible to avoid the unnecessary cast inside view holder's children by using a generic typing constrained by ViewBinding but somehow the compiler throws me a KotlinFrontEndException because of a ClassCastException so maybe it's actually not possible. Or maybe, if we're lucky, since they reworked casting in Kotlin 2.0, maybe they also touched some part in the involved casts that causes my issue? 

But anyway, I see no other solution to avoid the cast inside the child classes except maybe if we override the binding with the appropriate type inside each child with something link this:

```kt
override val binding: ItemMenuDrawerActionBinding = super.binding as ItemMenuDrawerActionBinding
```